recipes/Text_to_Shell/Text_to_Shell.ipynb

```diff
@@ -42,24 +57,132 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
    "metadata": {},
    "outputs": [],
    "source": [
-    "import ollama\n",
+    "import ollama"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's write two helper functions that we'll use for all our queries. First, we'll find it useful to determine the name of our operating system and use that string in queries. This is because shell commands sometimes have different options on Linux vs. MacOS, etc. We'll write our queries so they take this difference into account. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import platform\n",
     "\n",
-    "response = ollama.generate(\n",
-    "  model='granite-code:20b',\n",
-    "  prompt=\"How can I use a bash script to calculate the number of days between the dates Jan 12, 2023 and Jan 20, 2024?\"\n",
-    ")\n",
-    "print(response[\"response\"])"
+    "def os_name():\n",
+    "    os_name = platform.system()\n",
+    "    # It turns out, using \"MacOS\" is better than \"Darwin\", which is what gets returned on MacOS. \n",
+    "    # For all other cases, the returned value should be fine as is, so we map the result to the desired\n",
+    "    # name, but only for MacOS...\n",
+    "    name_map = {'Darwin': 'MacOS'}\n",
+    "    # ... then pass the os_name value as the second arg, which is used as the default return value.\n",
+    "    return name_map.get(os_name, os_name)  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "My OS is MacOS\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"My OS is {os_name()}\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "> **TIP:** If you are running on a Mac, it is likely that the invocation of `date` will be incorrect for the MacOS version of this command. Try adding the following sentence to the prompt and see what happens: `Make sure the generated shell commands are MacOS-compatible.`"
+    "Now let's write a helper function for running queries, wrapping the Ollama API. Note the default model name we'll used, Granite Code 20B (parameters). Also, note how we add additional content to the user's input prompt, such \"make sure you write code that works for _my_ system!\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def query(prompt: str, model: str = 'granite-code:20b') -> str:\n",
+    "    response = ollama.generate(\n",
+    "        model=model, \n",
+    "        prompt=f\"{prompt}. Make sure you generate code that is {os_name()}-compatible!\")\n",
```

Comment:
Could we put this in the system prompt rather than appending to the user prompt?

If we do that, we may not need the query wrapper. I would prefer not to bury the ollama.generate function if possible.

Reply:
I agree that's where it belongs. I followed you example of starting simple and adding more features, ultimately including a system prompt. I'll add a comment at that point that the system prompt is really where you want universal, repeated instructions like this.